src/p2p/pubsub.rs

```diff
@@ -182,22 +192,22 @@ impl Pubsub {
     /// SubscriptionStream.
     ///
     /// Returns true if an existing subscription was dropped, false otherwise
-    pub fn unsubscribe(&mut self, topic: impl Into<String>) -> bool {
-        let topic = Topic::new(topic);
-        if self.streams.remove(&topic).is_some() {
-            assert!(
-                self.floodsub.unsubscribe(topic),
-                "sender removed but unsubscription failed"
-            );
-            true
+    pub fn unsubscribe(&mut self, topic: impl Into<String>) -> Result<bool, Error> {
```

Comment:
Used `anyhow::Error` since i was unsure if I should passthrough `Gossipsub::unsubscribe` result, but even if I did it would still leave the matter of what to do for the other part of the if/else statement which is why I used `anyhow::Error`.